pkg/rc/farm.go

```diff
@@ -72,35 +81,14 @@ START_LOOP:
 	for {
 		select {
 		case <-quit:
-			rcf.logger.NoFields().Infoln("Halt requested, releasing replication controllers")
+			rcf.logger.NoFields().Infoln("Session expired, releasing replication controllers")
+			rcf.lock = nil
 			rcf.releaseChildren()
 			return
-		case session := <-rcf.sessions:
```

Comment:
There doesn't appear to be a way to get a new session ID without quitting this `for` loop. Is the implication that this function will be called once for every new session ID instead of forcing this function to handle the new session case?